src/app/workflow/launch-third-party/launch-third-party.component.html

```diff
@@ -1,22 +1,37 @@
-<div *ngIf="dnastackURL || fireCloudURL" class="panel panel-default">
+<div *ngIf="isWdl" class="panel panel-default">
   <div class="panel-heading">
     <h3>Launch with</h3>
   </div>
   <div class="panel-body">
     <div class="container-source-repos">
       <div class="container-launch-with">
-        <div class="button-wrap">
-          <div *ngIf="dnastackURL" class="button">
-            <p><a href="{{dnastackURL}}"><img src="../../assets/images/thirdparty/dnastack.png"> DNAstack &raquo;</a>
-            </p>
+        <div fxLayout="row wrap">
+          <div fxFlex="100%">
+            <div *ngIf="wdlHasFileImports || wdlHasHttpImports" class="import-warning">
+              Warning: this version of the WDL has imports, which are not supported by DNAstack. Make sure to select a version without imports in DNAstack.
+            </div>
+            <a mat-raised-button
+               title="{{!wdlHasContent ? 'The WDL has no content.' : ''}}"
+               target="_blank"
+               [attr.href]="dnastackURL"
+               [disabled]="!dnastackURL"
+               color="primary"><img src="../../assets/images/thirdparty/dnastack.png"> DNAstack &raquo;</a>
           </div>
-          <div *ngIf="fireCloudURL" class="button">
-            <p><a href="{{fireCloudURL}}"><img src="../../assets/images/thirdparty/FireCloud-white-icon.svg"> FireCloud
-              &raquo;</a></p>
+          <div fxFlex="100%">
+            <a mat-raised-button
+               title="{{!wdlHasContent ? 'The WDL has no content.' : (wdlHasFileImports ? 'FireCloud does not support file-path imports in WDL. It only supports http(s) imports.' : '')}}"
+               target="_blank"
+               [attr.href]="fireCloudURL"
+               [disabled]="!fireCloudURL"
+               color="primary"><mat-icon svgIcon="firecloud"></mat-icon> FireCloud &raquo;</a>
           </div>
-          <div *ngIf="dnanexusURL" class="button">
-            <p><a href="{{dnanexusURL}}"><img src="../../assets/images/thirdparty/DX_Logo_white_alpha.svg"> DNAnexus
-              &raquo;</a></p>
+          <div fxFlex="100%">
+            <a mat-raised-button
+               title="{{!wdlHasContent ? 'The WDL has no content.' : ''}}"
+               target="_blank"
```

Comment:
https://developers.google.com/web/tools/lighthouse/audits/noopener

Reply:
Thanks, I did not know about the `rel` attribute. Pushed a new commit with it.

Regarding the workflowsService subscribe errors, right now I'm initializing everything before making the http call, so no buttons will be enabled if there is an error.

More generally, I think we need a pattern/component for displaying errors for Ajax calls. Or maybe there is, and I'm unaware of it. I added it as a topic to discuss at today's meeting.